src/main.c

```diff
@@ -1007,23 +1008,41 @@ static void free_status_print(RaucStatusPrint *status)
 
 G_DEFINE_AUTOPTR_CLEANUP_FUNC(RaucStatusPrint, free_status_print);
 
+/* Definition list for terminal colors */
+#define KNRM  "\x1B[0m"
+#define KRED  "\x1B[31m"
+#define KGRN  "\x1B[32m"
+#define KYEL  "\x1B[33m"
+#define KBLU  "\x1B[34m"
+#define KMAG  "\x1B[35m"
+#define KCYN  "\x1B[36m"
+#define KWHT  "\x1B[37m"
+#define KBLD  "\x1B[1m"
+
 static void r_string_append_slot(GString *text, RaucSlot *slot, RaucStatusPrint *status)
 {
 	RaucSlotStatus *slot_state = slot->status;
 
-	g_string_append_printf(text, "\n  %s: class=%s, device=%s, type=%s, bootname=%s\n",
-			slot->name, slot->sclass, slot->device, slot->type, slot->bootname);
-	g_string_append_printf(text, "      state=%s, description=%s", r_slot_slotstate_to_str(slot->state), slot->description);
-	if (slot->parent)
-		g_string_append_printf(text, ", parent=%s", slot->parent->name);
-	else
-		g_string_append(text, ", parent=(none)");
-	if (slot->mount_point || slot->ext_mount_point)
-		g_string_append_printf(text, ", mountpoint=%s", slot->mount_point ? slot->mount_point : slot->ext_mount_point);
-	else
-		g_string_append(text, ", mountpoint=(none)");
+	setlocale(LC_ALL, "");
+
+	g_string_append_printf(text, "%lc %s%s%s"KBLD "[%s]"KNRM " (%s, %s, %s%s"KWHT ")"KNRM,
+			slot->bootname ? slot == status->primary ? L'\u23FA' :  L'\u25CB' : ' ',
+			slot->parent ? "  " : "",
+			slot->state & ST_ACTIVE ? KGRN : "",
+			slot->state == ST_BOOTED ? KGRN : "",
+			slot->name,
+			slot->device,
+			slot->type,
+			slot->state == ST_BOOTED ? KBLD : "",
+			r_slot_slotstate_to_str(slot->state));
+	if (slot->bootname)
+		g_string_append_printf(text, "\n\tbootname: "KBLU "%s"KNRM, slot->bootname);
+	if (slot->description)
```

Comment:
In you example you have empty description lines, maybe there is a pointer to an empty string?

Reply:
Yes, in config file parsing it is set to an empty string when being unset. I think we could simply drop this as we potentially handle a NULL description everywhere.